lib/Map/CesiumTileLayer.js

```diff
@@ -169,22 +170,26 @@ var CesiumTileLayer = L.TileLayer.extend({
         this._previousCredits = nextCredits;
     },
 
-    pickFeatures: function(map, longitudeRadians, latitudeRadians) {
+    pickFeatures: function(map, longitudeRadians, latitudeRadians, tileCoordinates, level) {
         var ll = new Cartographic(CesiumMath.negativePiToPi(longitudeRadians), latitudeRadians, 0.0);
-
-        var level = map.getZoom();
+        level = defaultValue(level, map.getZoom());
 
         var that = this;
         return pollToPromise(function() {
             return that.imageryProvider.ready;
         }).then(function() {
             var tilingScheme = that.imageryProvider.tilingScheme;
-            var tileCoordinates = tilingScheme.positionToTileXY(ll, level);
+            var tileCoordinates = defaultValue(tileCoordinates, tilingScheme.positionToTileXY(ll, level));
             if (!defined(tileCoordinates)) {
                 return undefined;
             }
 
-            return that.imageryProvider.pickFeatures(tileCoordinates.x, tileCoordinates.y, level, longitudeRadians, latitudeRadians);
+            return that.imageryProvider.pickFeatures(tileCoordinates.x, tileCoordinates.y, level, longitudeRadians, latitudeRadians).then(function(features) {
+                return {
+                    features: features,
+                    coords: {x: tileCoordinates.x, y: tileCoordinates.y, level: level}
+                };
+            });
```

Comment:
I'm just trying to understand the implications of this `then` clause... so `ImageryProvider.pickFeatures` says it returns a promise which resolves to an array of `ImageryLayerFeatureInfo` instances. So you're changing the result type of the `CesiumTileLayer.pickFeatures` promise... will it be confusing to have two different types of `pickFeatures` calls?

Is there any way to add some jsdoc-style explanation to this function?